ecosystem/indexer/src/models/token.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{models::events::Event, schema::tokens};
+use aptos_rest_client::types;
+use std::{collections::HashMap, fmt, fmt::Formatter, str::FromStr};
+
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(Associations, Debug, Identifiable, Insertable, Queryable, Serialize)]
+#[diesel(table_name = "tokens")]
+#[primary_key(token_id)]
+pub struct Token {
```

Comment:
Can we add `Clone` macros to these structs as well? If not, no worries

Reply:
added